### ISSUE TITLE
Consul - vault as a service mesh is broken in your Instruqt tutorial 

### ISSUE THREAD

**Issue by clumbo** [hashicorp/consul#17798]
#### Overview of the Issue

The "Configure Consul Connect service mesh CA in your Instruqt tutorial" is broken and I believe this to be a bug on a real deployed service too.

#### Reproduction Steps

1. Follow the tutorial here https://developer.hashicorp.com/consul/tutorials/vault-secure/vault-pki-consul-connect-ca
2. On the progress of the 3rd part when issuing this command

```
consul connect ca set-config -config-file ./assets/config-connect-ca-provider-vault.json`
```

it will fail with the following error:-

```
Error setting CA configuration: Unexpected response code: 500 (error configuring provider: Error making API request.

URL: GET http://vault.oyykoedjo9ge.svc.cluster.local:8200/v1/auth/token/lookup-self
Code: 403. Errors:

* permission denied)
```

**Comment by jkirschner-hashicorp** [comment]
Hi @clumbo,

Can you confirm the Vault policy associated with the Vault token or auth method in your Vault CA provider configuration?

The error message suggests that the Vault token or auth method lacks the `read` capability on `/auth/token/lookup-self`, but the tutorial documents the need for that permission in the Vault policy: https://developer.hashicorp.com/consul/tutorials/vault-secure/vault-pki-consul-connect-ca#create-vault-policies

![image](https://github.com/hashicorp/consul/assets/85913323/a41a432d-b08d-4e00-a4e6-404b1817361c)

**Comment by jkirschner-hashicorp** [comment]
EDIT: On first glance, I skipped over the mention of Instruqt. I have not yet checked whether Instruqt has an error that isn't present in the main tutorial text.

**Comment by jkirschner-hashicorp** [comment]
@clumbo : You are correct - Instruqt does not have an accurate Vault policy in step 2. It's missing `read` on `auth/token/lookup-self`, among other things. Thank you for bringing this to our attention!

![image](https://github.com/hashicorp/consul/assets/85913323/0a401839-082f-4019-8945-7927ebd26917)

---

In the meantime, try using these permissions instead from the main docs page (for Consul managed PKI paths - which is what Instruqt uses).

(Note: I changed the intermediate PKI path to `connect_inter` to match what Instruqt is using.)

```
path "/sys/mounts/connect_root" {
  capabilities = [ "create", "read", "update", "delete", "list" ]
}

path "/sys/mounts/connect_inter" {
  capabilities = [ "create", "read", "update", "delete", "list" ]
}

path "/sys/mounts/connect_inter/tune" {
  capabilities = [ "update" ]
}

path "/connect_root/*" {
  capabilities = [ "create", "read", "update", "delete", "list" ]
}

path "/connect_inter/*" {
  capabilities = [ "create", "read", "update", "delete", "list" ]
}

path "auth/token/renew-self" {
  capabilities = [ "update" ]
}

path "auth/token/lookup-self" {
  capabilities = [ "read" ]
}
```

**Comment by clumbo** [comment]
Thanks, I still have the same problem locally on a fresh install of vault

```
Error setting CA configuration: Unexpected response code: 500 (error configuring provider: Error making API request.

URL: PUT http://10.0.0.221:8200/v1/connect_dc1_inter/roles/leaf-cert
Code: 403. Errors:

* 1 error occurred:
	* permission denied)
```

**Comment by jkirschner-hashicorp** [comment]
In the Vault policy you configured, does it refer to the Vault intermediate path as `connect_inter` or `connect_dc1_inter`?

That error message suggests that Consul lacks the Vault permissions to call anything at `/connect_dc1_inter`. The corresponding Vault policy would look like:
```
path "/connect_inter/*" {
  capabilities = [ "create", "read", "update", "delete", "list" ]
}
```

If that says `path "/connect_dc1_inter/*"` instead, it doesn't work because the Vault policy doesn't grant permissions on the intermediate PKI path specified in Consul's Vault CA provider config.
